frontend/src/lib/components/AddInsightsToDashboard/addInsightsToDashboardModalLogic.ts

```diff
@@ -0,0 +1,101 @@
+import { dashboardLogic } from 'scenes/dashboard/dashboardLogic'
+import { savedInsightsLogic } from 'scenes/saved-insights/savedInsightsLogic'
+
+import { dashboardsModel } from '~/models/dashboardsModel'
+import { DashboardType, InsightModel, InsightShortId } from '~/types'
+
+import type { addInsightsToDashboardModalLogicType } from './addInsightsToDashboardModalLogicType'
+
+export interface AddInsightsToDashboardModalLogicProps {
+    dashboard: DashboardType
+}
+
+export interface Fuse extends FuseClass<any> {}
+
+export const addInsightsToDashboardModalLogic = kea<addInsightsToDashboardModalLogicType>([
+    props({} as AddInsightsToDashboardModalLogicProps),
+    key(({ dashboard }) => {
+        if (!dashboard.id) {
+            throw Error('must provide a dashboard id')
+        }
+        return dashboard.id
+    }),
+    path(['lib', 'components', 'AddInsightsToDashboard', 'saveInsightsToDashboardModalLogic']),
+    connect((props: AddInsightsToDashboardModalLogicProps) => ({
+        actions: [
+            dashboardLogic({ id: props.dashboard.id }),
+            ['addInsight', 'removeTile', 'refreshAllDashboardItems'],
+            eventUsageLogic,
+            ['reportSavedInsightToDashboard', 'reportRemovedInsightFromDashboard'],
+            dashboardsModel,
+            ['tileRemovedFromDashboard', 'tileAddedToDashboard'],
+        ],
+        values: [savedInsightsLogic, ['insights'], dashboardLogic({ id: props.dashboard.id }), ['tiles']],
+    })),
+    actions({
+        addNewInsight: true,
+        setInsightId: (short_id: InsightShortId) => ({ short_id }),
+        setSearchQuery: (query: string) => ({ query }),
+        setScrollIndex: (index: number) => ({ index }),
+        addToDashboard: (insight: InsightModel) => ({ insight }),
+        removeFromDashboard: (insight: InsightModel) => ({ insight }),
+    }),
+    reducers(({ props }) => ({
+        _insightId: ['', { setInsightId: (_, { short_id }) => short_id }],
+        searchQuery: ['', { setSearchQuery: (_, { query }) => query }],
+        scrollIndex: [-1 as number, { setScrollIndex: (_, { index }) => index }],
+        insightWithActiveAPICall: [
+            null as InsightShortId | null,
+            {
+                addToDashboard: (_, { insight }) => insight?.short_id || null,
+                removeFromDashboard: (_, { insight }) => insight?.short_id || null,
+                tileRemovedFromDashboard: (curr, { dashboardId }) => (dashboardId === props.dashboard.id ? null : curr),
+                tileAddedToDashboard: (curr, { dashboardId }) => (dashboardId === props.dashboard.id ? null : curr),
+            },
+        ],
+    })),
+    selectors({
+        insightId: [(s) => [s._insightId], (_insightId) => _insightId],
+        insightsFuse: [
+            (s) => [s.insights],
+            (insights): Fuse => {
+                return new FuseClass(insights.results || [], {
+                    keys: ['name', 'description', 'tags'],
```

Comment:
```suggestion
                    keys: ['name', 'derived_name', 'description', 'tags'],
```
This is necessary so that the automatically generated name is searched as well.